module-system/sov-modules-macros/src/rpc/expose_rpc.rs

```diff
@@ -0,0 +1,120 @@
+use proc_macro2::{Ident, Span};
+use quote::quote;
+
+pub(crate) fn expose_rpc(
+    args: proc_macro2::TokenStream,
+    input: syn::ItemImpl,
+) -> Result<proc_macro::TokenStream, syn::Error> {
+    let attrs = &input.attrs;
+
+    let args: syn::Type = syn::parse2(args).expect("Expected a valid type list");
+
+    let mut output_tokens = proc_macro2::TokenStream::new();
+
+    let mut merge_operations = proc_macro2::TokenStream::new();
+
+    let types = match args {
+        syn::Type::Tuple(tuple) => tuple.elems,
+        _ => panic!("Expected a tuple of types"),
+    };
+
+    let rpc_storage_struct = quote! {
+        #[derive(Clone)]
+        pub struct RpcStorage<C: Context> {
```

Comment:
I don't know if it is related to this PR, but shouldn't context to be fully qualified?
```suggestion
        pub struct RpcStorage<C: ::sov_modules_api::Context> {
```